source/tutorials/installing-nixos-on-a-raspberry-pi.rst

```diff
@@ -0,0 +1,184 @@
+Installing NixOS on a Raspberry Pi
+==================================
+
+This tutorial assumes `Raspberry P 4 Model B with 4GB RAM <https://www.raspberrypi.org/products/raspberry-pi-4-model-b/>`_.
+
+Before starting this tutorial, make sure you have 
+`all necessary hardware <https://projects.raspberrypi.org/en/projects/raspberry-pi-setting-up/1>`_:
+
+- HDMI cable/adapter.
+- 8GB+ SD card.
+- SD card reader in case your machine doesn't have an SD slot.
+- Power cable for your Raspberry Pi.
+- USB keyboard.
+
+.. note:: 
+
+  This tutorial was written for the Raspberry Pi 4B. Using a previous supported revision, like the 3B or 3B+, is possible with tweaks.
+
+
+Booting NixOS live image
+------------------------
+
+.. note:: Booting from USB may require an EEPROM firmware upgrade. This tutorial boots from an SD card to avoid such hiccups.
+
+Prepare the AArch64 image on your laptop:
+
+.. code:: shell-session
+
+  $ nix-shell -p wget zstd
+  $ wget https://hydra.nixos.org/build/142828023/download/1/nixos-sd-image-21.05pre288297.8eed0e20953-aarch64-linux.img.zst
+  $ unzstd -d nixos-sd-image-21.05pre288297.8eed0e20953-aarch64-linux.img.zst
+  $ dmesg --follow
+
+Your terminal should be printing kernel messages as they come in.
+
+Plug in your SD card and your terminal should print what device it got assigned, for example ``/dev/sdX``.
+
+Press ``ctrl-c`` to stop ``dmesg -w``.
+
+Copy NixOS to your SD card by replacing ``sdX`` with the name of your device:
+
+.. code:: shell-session 
+
+  sudo dd if=nixos-sd-image-21.05pre288297.8eed0e20953-aarch64-linux.img of=/dev/sdX bs=4096 conv=fsync status=progress
+
+Once that command exits, **move the SD card into your Raspberry Pi and power it on**.
+
+You should be greeted with a fresh shell!
+
+
+Getting internet connection
+---------------------------
+
+Run ``sudo -i`` to get a root shell for the rest of the tutorial.
+
+At this point we'll need internet connection. If you can use an ethernet cable, plug it in.
+
+In case you're connecting to a wifi run ``iwconfig`` to see what is the name of your wireless
+network interface. In case it's ``wlan0`` replace ``SSID`` and ``passphrase`` with your data and run:
+
+.. code:: shell-session 
+
+  # wpa_supplicant -B -i wlan0 -c <(wpa_passphrase 'SSID' 'passphrase') &
+
+
+Once you see in your terminal that connection is established, run ``host google.com`` to 
```

Comment:
`google.com` is not a connection testing service!

If we are to namedrop a domain name, it could also be `nixos.org`